src/appshell/view/dockwindow/internal/dockbase.cpp

```diff
@@ -640,6 +642,12 @@ void DockBase::updateFloatingStatus()
     doSetFloating(floating);
 }
 
+void DockBase::onIsInMainWindowChanged()
+{
+    applySizeConstraints();
+    updateFloatingStatus();
```

Comment:
Is this method executed only when `isInMainWindow` has changed to `true`, or also when it has changed to `false`? If the latter, isn't this duplicating the code in line +621 to +624, or is that code invoked in different situations?

Or, to rephrase my question: when is `KDDockWidgets::Frame::isInMainWindowChanged` emitted and when `KDDockWidgets::DockWidgetQuick::parentChanged`?

Reply:
>Is this method executed only when isInMainWindow has changed to true, or also when it has changed to false? 

It is called in both cases

>If the latter, isn't this duplicating the code in line +621 to +624, or is that code invoked in different situations?

The code in lines 621-624 is needed for one particular case: when the dock becomes undocked and loses its origin frame. KDDockWidgets::Frame::isInMainWindowChanged won't be called in this case (we just haven't yet connected to this signal from a new frame). This code ensures that the size constraints will be applied to the floating dock

Reply:
PS: docks do not always change their parent when they are undocked. Several docks can use the same frame at the same time. And that frame really belongs to only one of them (the first one who created it)

Reply:
Ah, thanks for the clarification!